python/cudf/cudf/_lib/text.pyx

```diff
@@ -31,9 +33,25 @@ def read_text(object filepaths_or_buffers,
 
     cdef unique_ptr[data_chunk_source] datasource
     cdef unique_ptr[column] c_col
-
-    with nogil:
-        datasource = move(make_source_from_file(filename))
-        c_col = move(multibyte_split(dereference(datasource), delim))
+    cdef size_t c_byte_range_offset
+    cdef size_t c_byte_range_size
+    cdef byte_range_info c_byte_range
```

Comment:
Is there any reason we want to declare these always rather than only when `byte_range is not None`?

Reply:
cython didn't allow me to put cdef inside of a conditional branch.

Reply:
Cython will not allow `cdef` in conditional blocks. The semantics of how that should translate into C(++) code are ambiguous in a lot of cases. There's a discussion of this somewhere on the cython-users mailing list that I can dig up if you're curious.